lca-ai-stack/source/lambda_functions/call_event_processor/event_processor/call_event_processor.py

```diff
@@ -1539,33 +1569,17 @@ async def execute_process_event_api_mutation(
             return_value["errors"].append(response)
         else:
             return_value["successes"].append(response)
-    elif('Metadata' in message.keys()):
+    elif event_type == "CALL_ANALYTICS_METADATA":
         meta = json.loads(message['Metadata'])
         LOGGER.debug("S3 URL from metadata %s", meta['oneTimeMetadata']['s3RecordingUrl'])
 
         bucket = meta['oneTimeMetadata']['s3RecordingUrl'].split("/")[5].split("?")[0]
         region = re.search(r"region\=(.+)\&prefix+", meta['oneTimeMetadata']['s3RecordingUrl']).group(1)
-        prefix=re.search(r"\&prefix\=(.+)", meta['oneTimeMetadata']['s3RecordingUrl']).group(1)
+        prefix = re.search(r"\&prefix\=(.+)", meta['oneTimeMetadata']['s3RecordingUrl']).group(1)
+        session_id = re.search(r"\/(.+\/)*(.+)\.(?i)(wav)$", meta['oneTimeMetadata']['s3RecordingUrl']).group(2)
```

Comment:
In python3.11 this throws error; `global flags not at the start of the expression`
Correct syntax is
`session_id = re.search(r"(?i)\/(.+\/)*(.+)\.(wav)$", meta['oneTimeMetadata']['s3RecordingUrl']).group(2)` - moving (?i) to start of the expression.

Reply:
Fixed in develop branch - [commit](https://github.com/aws-samples/amazon-transcribe-live-call-analytics/commit/5dfc5b3e58befd3e777eae795ceb4f6e2cc04e51)